packages/server/src/arpa_reporter/lib/audit-report.js

```diff
@@ -208,17 +136,69 @@ async function createReportsGroupedByProject(periodId) {
     const recordsByProject = getRecordsByProject(records);
     const reportingPeriods = await getAllReportingPeriods();
 
-    const inputs = [];
+    return Object.entries(recordsByProject).map(([projectId, projectRecords]) => {
+        const record = projectRecords[0];
+
+        // set values for columns that are common across all records of projectId
+        const row = {
+            'Project ID': projectId,
+            'Project Description': record.content.Project_Description__c,
+            'Project Expenditure Category Group': ec(record.type),
+            'Project Expenditure Category': record.subcategory,
+        };
+
+        // get all reporting periods related to the project
+        const allReportingPeriods = Array.from(new Set(projectRecords.map((r) => r.upload.reporting_period_id)));
+
+        // initialize the columns in the row
+        allReportingPeriods.forEach(async (reportingPeriodId) => {
```

Comment:
Does this need async?

Reply:
It does not at all.  Thank you, great catch!